scripts/ci/preflight-steps/dependency_updater.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python3
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+import re
+import requests
+
+class VersionUpdater:
+    def __init__(self, file, dependency):
+        self.file = file
+        self.dependency = dependency
+        self.dependencies = {}
+
+    def get_latest_version(self, org_name, crate_name):
+        url = f"https://github.com/{org_name}/{crate_name}/releases/latest"
+        response = requests.get(url)
+        if response.status_code == 404:
+            raise ValueError(f"Failed to find latest version for crate '{crate_name}' on GitHub.")
+        else:
+            return re.search(r"v?(\d+\.\d+(\.\d+)?(-\S+)?)", response.url).group(1)
+
+    def read_dependencies(self):
+
+        with open(self.file, 'r') as f:
+            contents = f.readlines()
+
+        for line in contents:
+            if "CBMC_VERSION" in line:
+                version_number = line.split("=")[1].replace("\n","")
+                self.dependencies["CBMC_VERSION"] = version_number
+            elif "CBMC_VIEWER_VERSION" in line:
+                version_number = line.split("=")[1].replace("\n","")
+                self.dependencies["CBMC_VIEWER_VERSION"] = version_number
+            elif "KISSAT_VERSION" in line:
+                version_number = line.split("=")[1].replace("\n","")
+                self.dependencies["KISSAT_VERSION"] = version_number
+            else:
+                pass
+
+    def update_dependencies(self):
+        latest_version = self.get_latest_version(self.dependency["org_name"], self.dependency["dependency_name"])
+        self.dependencies[self.dependency["dependency_string"]] = latest_version
+
+    def write_dependencies(self):
+        with open(self.file, 'r') as f:
+            lines = f.readlines()
+        with open(self.file, 'w') as f:
+            for line in lines:
+                match = re.search(r'(\w+)="\d+(\.\d+)+"', line.strip())
+                if match and match.group(1).strip() == self.dependency["dependency_string"]:
+                    f.write(f'{self.dependency["dependency_string"]}="{self.dependencies[self.dependency["dependency_string"]]}"\n')
+                else:
+                    f.write(line)
+
+
+    def run_process(self):
```

Comment:
It's not clear to me that you need to walk through the file, updating the existing values. It seems much quicker to overwrite the entire file.

If you have a dict like this:
```pr
new_versions = {
  "CBMC_VIEWER": 3.9,
  "CBMC": 5.99.0,
  ...
}
```
then I think it would be much clearer to do this:
```
with open("kani-dependencies", "w") as handle:
    for tool, version in new_versions.items():
        print(f"{tool}={version}", file=handle)
```
then you don't need the if-statement on line 27, which hard-codes the tool names and will thus need to be updated whenever we add a new dependency.

so my suggestion is:
1. Get the list of existing tools by opening `kani-dependencies`, read each line, split on `=`, return the first part
2. Turn that list of tools into the `new versions` dict, using a variation of your `get_latest_version` method
3. Write the dict out using the 3 lines of code above, no need to update the existing file, just overwrite it.

I think this would make the code a lot shorter and clearer, and avoid hardcoding the tool names here. `kani-dependencies` is the only place those names should be written.